### ISSUE TITLE
DeprecationWarning: generator ... raised StopIteration

### ISSUE THREAD

**Issue by samuelkolb** [pysmt/pysmt#488]
FYI, I'm getting a DepreciationWarning in two places when using`pysmt.shortcuts.read_smtlib()`:

```
.../env/lib/python3.6/site-packages/pysmt/smtlib/parser.py:179: DeprecationWarning: generator 'Tokenizer.create_generator' raised StopIteration
  return next(self.generator)
```
```
.../env/lib/python3.6/site-packages/pysmt/smtlib/parser.py:796: DeprecationWarning: generator 'SmtLibParser.get_command' raised StopIteration
  for cmd in self.get_command(tokens):
```

**Comment by samuelkolb** [comment]
Okay, thanks for the information, I'll try to have a look at this soon.

**Comment by marcogario** [comment]
I am unable to reproduce.

I think this has to do with SMT-LIB files with annotations, but I tried a few that we have in our test-set and none seems to raise this exception.  Could you please submit an example of this?

**Comment by marcogario** [comment]
This should be fixed once we merge #521.

**Comment by samuelkolb** [comment]
Sorry for dropping the ball on this - if I understand correctly this issue will be fixed in release 0.7.6 but that release is not ready yet? Because under Python 3.7 this issue breaks reading from smt_lib files as the stop iteration is no longer silenced.

**Comment by samuelkolb** [comment]
Steps to reproduce the error:
- run python 3.7
- download (as file.smt2) https://clc-gitlab.cs.uiowa.edu:2443/SMT-LIB-benchmarks/QF_LRA/raw/master/2017-Heizmann-UltimateInvariantSynthesis/_array1.i_3_2_2.bpl_11.smt2
- pysmt.shortcuts.read_smtlib(file.smt2)
